src/native/libs/System.Security.Cryptography.Native/opensslshim.c

```diff
@@ -233,9 +233,9 @@ void InitializeOpenSSLShim(void)
     }
 
 #if defined(TARGET_ARM) && defined(TARGET_LINUX)
-    // This value will represent a time in year 2038 if 64-bit time is used,
-    // or 1901 if the lower 32 bits are interpreted as a 32-bit time_t value.
-    time_t timeVal = (time_t)INT_MAX + 1;
+    // This value will represent a time in year 2038 if 64-bit time_t is used,
+    // or 1901 if the conversion overflowed.
+    time_t timeVal = (time_t)0x80000000U;
```

Comment:
Okay. Now I understand the intention behind this check.

> The logic was assuming we use 64-bit time_t.

Is that a reasonable assumption? Should we just set `g_libSslUses32BitTime` to `1` if `sizeof(time_t) == sizeof(int32_t)` and not bother asking OpenSSL? 

> Could we instead fix this by updating the armv6 build to use 64-bit time_t?

That would perhaps work for our CI, but what about other community builds where time_t is 32-bit?

Reply:
> I think it's possible we built with 32-bit time_t, but are running on a system where openssl uses 64-bit time_t.

But that does not work right now, either, since you said "The logic was assuming we use 64-bit time_t." If we require a 64-bit time_t then we static assert it so folks have a better compilation message.


```C
static_assert(sizeof(time_t) == 8, "64-bit time_t is required.");
```

If we do that, then we can still switch to using `(time_t)0x80000000` so that people using a 32-bit time_t do not see error messages around overflows.

That doesn't help us with what to do about the armv6 build. Are we basically saying that .NET _requires_ a 64-bit time_t now?

> or do the community builds need to continue targeting 32-bit time_t?

I don't know much about those builds or who owns them. @ViktorHofer or @akoeplinger may know?

Reply:
Okay. I added a static assert and undid some comment changes. I think replacing INT_MAX + 1 is still helpful so that it avoids UB on systems where time_t is 32-bit, that way the only error they see is the static assert when building.

This puts us back in to a broken build, but at least now the build is broken with a better error message.